### PR TITLE
Maven Central Migration - Library updates to fix build

### PR DESCRIPTION
**Pull request by lauren2020**
Issue: #768 

This PR updates outdated dependencies that were preventing the project from building and being moved to maven central.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -109,21 +103,32 @@ allprojects {
...
-        classpath "com.github.dcendents:android-maven-gradle-plugin:2.1"
-        classpath "com.jfrog.bintray.gradle:gradle-bintray-plugin:1.8.4"
-        classpath "org.jfrog.buildinfo:build-info-extractor-gradle:4.8.1"
-        classpath "com.vanniktech:gradle-android-javadoc-plugin:0.3.0"
-        classpath "de.mannodermaus.gradle.plugins:android-junit5:1.3.2.0"
-        classpath "eu.davidea:grabver:1.0.1"
+        classpath "com.android.tools.build:gradle:$gradle_plugin_version"
+        classpath "com.neenbedankt.gradle.plugins:android-apt:$android_apt_version"
+        classpath "com.github.dcendents:android-maven-gradle-plugin:$maven_gradle_plugin_version"
+        classpath "org.jfrog.buildinfo:build-info-extractor-gradle:$build_info_extractor_gradle_version"
```
Comment (barnhill) [inline build.gradle:128]: Jfrog is JCenter so we should try to remove all their stuff

File: flexible-adapter/build.gradle [hunk 2]
```diff
@@ -49,6 +49,6 @@ dependencies {
 if (project.hasProperty('all') || project.hasProperty('adapter')) {
     apply from: '../maven-install.gradle'
     apply from: '../maven-publish.gradle'
-    apply from: '../jfrog-bintray-publish.gradle'
+    //apply from: '../jfrog-bintray-publish.gradle'
```
Comment (barnhill) [inline flexible-adapter/build.gradle:52]: Remove this as it's JCenter

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.3.0-all.zip
+distributionUrl=https://services.gradle.org/distributions/gradle-6.7.1-all.zip
```
Comment (barnhill) [inline gradle/wrapper/gradle-wrapper.properties:6]: 6.8.3 is out

File: build.gradle [hunk 4]
```diff
@@ -15,49 +15,42 @@ allprojects {
...
-                commonJava8    : "android.arch.lifecycle:common-java8:${archExtensionsVer}",
-                reactiveStreams: "android.arch.lifecycle:reactivestreams:${archExtensionsVer}",
-                coreTesting    : "android.arch.core:core-testing:${archExtensionsVer}",
-                room           : "android.arch.persistence.room:runtime:${roomPersistenceVer}",
-                roomCompiler   : "android.arch.persistence.room:compiler:${roomPersistenceVer}",
-                roomRxJava2    : "android.arch.persistence.room:rxjava2:${roomPersistenceVer}",
-                navigation     : "android.arch.navigation:navigation-fragment:${navigationVer}"
-        ]
+        targetSdk = 30 // Android 11
+        buildTools = "29.0.2"
```
Comment (barnhill) [inline build.gradle:19]: This is no longer needed as it's included in the Gradle plugin now

File: build.gradle [hunk 5]
```diff
@@ -109,21 +103,32 @@ allprojects {
 
 buildscript {
     repositories {
-        jcenter()
+        mavenCentral()
+        gradlePluginPortal()
         google()
         maven { url "http://dl.bintray.com/davideas/maven" }
```
Comment (barnhill) [inline build.gradle:109]: This is a JCenter url.

**Review by barnhill — CHANGES_REQUESTED** [review]

File: flexible-adapter-app/src/main/res/layout/fragment_recycler_view_data.xml [hunk 6]
```diff
@@ -31,8 +31,8 @@
                     android:id="@+id/recycler_view"
                     android:layout_width="match_parent"
                     android:layout_height="match_parent"
-                    app:items="@{items}"
                     tools:listitem="@layout/recycler_simple_item" />
+                <!-- app:items="@{items}" was causing build error, temporarily removed -->
```
Comment (lauren2020) [inline flexible-adapter-app/src/main/res/layout/fragment_recycler_view_data.xml:35]: This was causing a build error for the demo app, would like to move fix for demo app to a separate pr to not hold up general build fix for maven central migration

**Comment by barnhill** [comment]
gradlepluginportal() may point to JCenter too. We need to eval if this is necessary.

**Pull request merged**